Dalamud/Interface/Internal/Windows/PluginInstallerWindow.cs

```diff
@@ -1512,18 +1563,18 @@ private void DrawVisitRepoUrlButton(string? repoUrl)
             }
         }
 
-        private bool DrawPluginImages(PluginManifest manifest, int index, bool isThirdParty)
+        private bool DrawPluginImages(LocalPlugin? plugin, PluginManifest manifest, bool isThirdParty, int index)
         {
-            if (!this.pluginImagesMap.TryGetValue(manifest.InternalName, out var images))
+            var hasImages = this.pluginImagesMap.TryGetValue(manifest.InternalName, out var imageTextures);
+            if (!hasImages)
             {
-                Task.Run(() => this.DownloadPluginImagesAsync(manifest, isThirdParty));
-                return false;
-            }
+                this.pluginImagesMap.Add(manifest.InternalName, Array.Empty<TextureWrap>());
+                Task.Run(async () => await this.DownloadPluginImagesAsync(plugin, manifest, isThirdParty));
 
-            if (!images.IsDownloaded)
                 return false;
+            }
 
-            if (images.Textures == null)
+            if (imageTextures.Length == 0)
```

Comment:
the images array will never be null, either an empty array, or an array with possible nulls in it.